test/integration/src/test/java/com/liferay/faces/portal/test/integration/demo/PrimeFacesUsersPortletTester.java

```diff
@@ -48,506 +52,621 @@
+		BrowserDriver browserDriver = getBrowserDriver();
 
-		// Verify that when "Adams" is entered into the last name filter that only users with the last name "Adams"
-		// are present in the list.
-		String lastNameColumnFilterXpath = "//input[contains(@id,'users:lastName:filter')]";
-		Actions filterActions = browserDriver.createActions(lastNameColumnFilterXpath);
-		WebElement lastNameColumnFilterElement = browserDriver.findElementByXpath(lastNameColumnFilterXpath);
-		filterActions.sendKeys(lastNameColumnFilterElement, "Adams");
-		browserDriver.performAndWaitForRerender(filterActions.build(), "(//span[contains(@id,':firstNameCell')])[1]");
+		// 1. Take note of each screen name in the first column of the table, clicking the *Next* button until all
+		// the screen names have been noted.
+		List<String> screenNames = extractColumnValuesFromDataTable(browserDriver, "screenName");
+		logList(screenNames, "Screen names extracted from the UI:");
 
-		List<String> lastNames = extractColumnValuesFromDataTable(browserDriver, "lastName");
-		Assert.assertTrue(lastNames.size() > 1);
+		// 2. Sort the initial unsorted list of screen names to obtain the expected list sorted in ascending order.
+		List<String> expectedScreenNamesAscending = new ArrayList<String>(screenNames);
+		Collections.sort(expectedScreenNamesAscending);
+		logList(expectedScreenNamesAscending, "Expected sorted list of screen names (ascending):");
 
-		for (String lastName : lastNames) {
-			Assert.assertTrue("Adams".equals(lastName));
-		}
+		// 3. Click the *Screen Name* header to sort screen names into ascending order and verify that the sort
+		// indication icon is pointed up.
+		browserDriver.clickElement(SCREEN_NAME_SORT_BUTTON_XPATH);
 
-		// Clear the filter and verify that more than one user is present in the list of users.
-		browserDriver.clearElement(lastNameColumnFilterXpath);
-		browserStateAsserter.assertElementDisplayed(SECOND_USER_SCREEN_NAME_CELL_XPATH);
+		BrowserStateAsserter browserStateAsserter = getBrowserStateAsserter();
+		browserStateAsserter.assertElementDisplayed(SORTED_ASCENDING_ICON_XPATH);
+
+		// 4. Take note of each screen name in the first column of the table, clicking the *Next Page* button until
+		// all the screen names have been noted.
+		List<String> actualScreenNamesAscending = extractColumnValuesFromDataTable(browserDriver, "screenName");
+		logList(actualScreenNamesAscending,
+			"Actual screen names extracted from the UI after sorting by ascending order:");
+
+		// 5. Verify that the sorted list obtained from the UI matches the order of the expected list (sorted in
+		// ascending order).
+		Assert.assertEquals(expectedScreenNamesAscending, actualScreenNamesAscending);
+
+		// 6. Click the *First Page* button to go back to the first page of users.
+		browserDriver.clickElement(getNavigationButtonXpath("First"));
+		browserDriver.waitFor(pageButtonClassActive(1));
```

Comment:
@stiemannkj1 Not sure it would be necessary to append the word "condition" here, since it reads nicely enough as it is, but you could consider it for the sake of consistency.